### ISSUE TITLE
[Bug] Weapons sometimes not firing but drawing power/making heat [20w41c]

### ISSUE THREAD

**Issue by Dwarf-LordPangolin** [tsunamayo/Starship-EVO#3061]
### Short Description of the issue:
Weapons sometimes do not fire, but will still consume power and generate heat.

### Steps to reproduce the issue:
Not sure how to reproduce it, so I'm sending a blueprint that has this problem. Just hop in and fire the weapons enough times, it should show up sooner or later.

### Environment
- Starship EVO version: 20w41c
- Graphics Card: NVIDIA GeForce GTX 1060 6GB

### Additional information:
[output_log.txt](https://github.com/tsunamayo/Starship-EVO/files/5360577/output_log.txt)

[Mako Light Fighter 3.zip](https://github.com/tsunamayo/Starship-EVO/files/5360578/Mako.Light.Fighter.3.zip)

**Comment by ProPeach** [comment]
I may be wrong, but I believe this happens because weapons draw energy/produce heat when they're recharging rather than when they actually fire. So if your weapons produce too much heat for your ship to dissipate, they take a few false starts to charge back up before firing again as the heat limit is hit and the ship has to cool down before starting to charge again.

I think there are a few other factors which add to the confusion too, like the fact that the heat bar seems to have a maximum speed it can move at - even if you go from 0-100% heat in an instant, it take a while for the heat bar to show this. It's easier to see with beams or large lasers.

To remedy this, weapons should draw power/produce heat when they *fire*, not when they *recharge*. That, combined with a heat bar that reacts immediately to heat changes would be much clearer to the player in indicating what exactly is happening to their ship as at the moment it's not very intuitive

**Comment by Dwarf-LordPangolin** [comment]
Ahh, I see! I tested it more closely, and you're right. So if the ship reaches the heat cap while the beams are recharging & cooling, the beams "pause" what they're doing. Firing them afterwards just unpauses the cooling and starts it again, and if the heat is too high it might end up with another pause or more before they finish cooling down and can actually fire again.

**Comment by tsunamayo** [comment]
Yeah guy sorry I am being busy on interstellar travel. I think peach is right, at least yes the heat draw is during the charging period. If it is confusing I will change it!
